std/dynamic_library.zig

```diff
@@ -261,20 +262,58 @@ pub const WindowsDynLib = struct {
     }
 
     pub fn lookup(self: *WindowsDynLib, name: []const u8) ?usize {
-        return @ptrToInt(windows.kernel32.GetProcAddress(self.dll, name.ptr));
+        if (windows.kernel32.GetProcAddress(self.dll, name.ptr)) |addr| {
+            return @ptrToInt(addr);
+        } else {
+            return null;
+        }
+    }
+};
+
+pub const DarwinDynLib = struct {
+    handle: *c_void,
+
+    pub fn open(path: []const u8) !DarwinDynLib {
+        // should we perform this check?
```

Comment:
I don't think we need this: if `dlopen` fails it will let us know

Reply:
but getting the actual error isn't going to be that easy